package.json

```diff
@@ -47,6 +47,33 @@
       "title": "Show message when a file is ignored",
       "type": "boolean",
       "default": false
+    },
+    "grammarScopes": {
+      "title": "Enable for grammars",
+      "type": "array",
+      "default": [
+        "source.css",
+        "source.scss",
+        "source.css.scss",
+        "source.less",
+        "source.css.less",
+        "source.css.postcss",
+        "source.css.postcss.sugarss",
+        "source.inside-js.css.styled"
+      ],
+      "items": {
+        "type": "string",
+        "enum": [
```

Comment:
Wow, although this _works_, it leads to some incredibly broken user experiences. Typing anything other than this allowed list in there results in that non-allowed item disappearing with no warning... this includes accidentally editing one of the allowed list.

There are two approaches that can be taken here:
* Have a predefined list of scopes, with individual options controlling each one, similar to how https://github.com/AtomLinter/linter-stylelint/commit/c9da67a6fa19c9961ff8ee32feb7bc938342de27 worked.
* Allow any scopes to be entered, similar to how `linter-eslint` works ([1](https://github.com/AtomLinter/linter-eslint/blob/v8.1.2/package.json#L75), [2](https://github.com/AtomLinter/linter-eslint/blob/v8.1.2/src/main.js#L29))

Since this package currently can only use the embedded `stylelint`, I would say that the first option is the better way to go for now as extra processors currently can't be added by the user (without editing the source code)

tl;dr: Change this whole option to an "Enable Styled Components Support" option 😛.